source/common/stats/symbol_table.cc

```diff
@@ -640,6 +640,11 @@ const uint8_t* StatNamePool::addReturningStorage(absl::string_view str) {
   return storage_vector_.back().bytes();
 }
 
+StatName StatNamePool::add(StatName name) {
+  storage_vector_.push_back(Stats::StatNameStorage(name, symbol_table_));
```

Comment:
qq: should this be emplace_back for maximum perf?

I get confused when emplace_back should be used and this seems like it might be one of those.

Reply:
yes, I think this should be `emplace` instead, changed it in both places to use emplace so we only create the object in the container.